generators/app/templates/infrastructure/src/components/menu/MenuItem.js

```diff
@@ -1,42 +1,56 @@
 import React from "react";
 import PropTypes from "prop-types";
 import cx from "classnames";
-import { ListItem, ListItemIcon, ListItemText, makeStyles, Tooltip } from "@material-ui/core";
+import { ListItem, ListItemIcon, ListItemText, ListItemButton, makeStyles, Tooltip } from "@material-ui/core";
+import { ArrowDropDown, ArrowDropUp } from "@material-ui/icons";
 import { NavLink } from 'react-router-dom';
 import menuStyle from "assets/jss/components/menuStyle";
 import { useTranslation } from 'react-i18next';
+import { isNil } from "ramda";
 
 const useStyles = makeStyles(menuStyle);
 
-const MenuItem = ({ menu, drawerOpen, activeRoute }) => {
-  const classes = useStyles();
+const MenuItem = ({ menu, drawerOpen, activeRoute, isSubMenuItem, subMenuOpen, onToggleSubMenu }) => {
+  const { children, path, icon, text } = menu;
+  const isSubMenu = !isNil(children);
+
+  const classes = useStyles({ isSubMenu });
   const { t } = useTranslation();
+
   const navLinkClasses = classes.menuItemLink +
     " " +
     cx({
-      [" " + classes.menuActiveColor]: activeRoute(menu.path)
+      [classes.menuActiveColor]: !isSubMenu && activeRoute(path)
     });
-  const itemText = classes.menuItemText +
+  const itemTextClasses = classes.menuItemText +
     " " +
     cx({
       [classes.menuItemTextMini]: !drawerOpen
+    }) +
+    " " +
+    cx({ 
+      [classes.paddingLeft]: isSubMenuItem 
     });
+  const menuItemIconClasses = classes.menuItemIcon + ' ' + cx({ [classes.paddingLeft]: isSubMenuItem });
 
-  const text = t(menu.text)
+  const translatedText = t(text)
+  const Item = isSubMenu ? ListItemButton : NavLink;
+  const itemProps = isSubMenu ? { onClick: onToggleSubMenu } : { to: path };
 
   return (
-    <Tooltip disableHoverListener={!drawerOpen} title={text}>
+    <Tooltip disableHoverListener={!drawerOpen} title={translatedText}>
       <ListItem className={classes.menuItem}>
-        <NavLink to={menu.path} className={navLinkClasses}>
-          <ListItemIcon className={classes.menuItemIcon}>
-            {menu.icon}
+        <Item {...itemProps} className={navLinkClasses}>
+          <ListItemIcon className={menuItemIconClasses}>
+            {icon}
           </ListItemIcon>
           <ListItemText
-            primary={text}
+            primary={translatedText}
+            secondary={isSubMenu && (subMenuOpen ? <ArrowDropUp className={classes.caret} /> : <ArrowDropDown className={classes.caret} />)}
```

Comment:
what happens when this is false? did you check the console for any warnings?

Reply:
it doesn't cause any warning, ListItemText receives undefined for secondary prop and doesn't draw anything in that place

Reply:
are you sure? If **isSubMenu** is **false** I think the **secondary** prop will receive **false** not **undefined**. And **secondary** prop only accepts a **node** type value right?

Reply:
I was wrong, it's false :d but it seems boolean is considered to be a react node type that is ignored and this is why no warning is thrown and nothing is drawn for secondary in the UI
https://reactnative.dev/docs/react-node